File: Java/StopThread.java
import java.util.Arrays;
import java.util.Comparator;

public class StopThread extends Thread {
    private final StartThread[] threads;

    public StopThread(StartThread[] threads) {
        this.threads = Arrays.stream(threads)
                .sorted(Comparator.comparingInt(StartThread::getMillis))
                .toArray(StartThread[]::new);
    }

    @Override
    public void run() {
        for (int i = 0; i < threads.length; i++) {
            try {
                int sleepTime;

                if (i == 0) {
                    sleepTime = threads[i].getMillis();
                } else {
                    sleepTime = threads[i].getMillis() - threads[i - 1].getMillis();
                }

                Thread.sleep(sleepTime);

                threads[i].running = false;
                System.out.println("Stopped thread " + threads[i].getID());
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }
}

File: C#/StopThread.cs
using System.Linq;
using System.Threading;

namespace LB1
{
    class StopThread
    {
        private StartThread[] threads { get; set; }

        public StopThread(StartThread[] threads, int[] delays)
        {
            this.threads = threads.OrderBy(t => t.Millis).ToArray();
        }

        public void Breaker()
        {
            for (int i = 0; i < threads.Length; i++)
            {
                int sleepTime;

                if (i == 0)
                {
                    sleepTime = threads[i].Millis;
                }
                else
                {
                    sleepTime = threads[i].Millis - threads[i - 1].Millis;
                }

                Thread.Sleep(sleepTime);
                threads[i].Running = false;
            }
        }
    }
}
